lib/mayaUsd/ufe/UsdUndoRenameCommand.cpp

```diff
@@ -88,84 +89,85 @@ UsdUndoRenameCommand::~UsdUndoRenameCommand()
+        }
 
         if (status) {
             // The renamed scene item is a "sibling" of its original name.
-            auto ufeSrcPath = _ufeSrcItem->path();
-            _ufeDstItem = createSiblingSceneItem(
-                ufeSrcPath, _usdDstPath.GetElementString());
+            _ufeDstItem = createSiblingSceneItem(_ufeSrcItem->path(), _usdDstPath.GetElementString());
 
-            Ufe::ObjectRename notification(_ufeDstItem, ufeSrcPath);
-            Ufe::Scene::notifyObjectPathChange(notification);
+            sendRenameNotification(_ufeDstItem, _ufeSrcItem->path());
         }
     }
     else {
         UFE_LOG(std::string("Warning: SdfCopySpec(") +
                 _usdSrcPath.GetString() + std::string(") failed."));
     }
 
-	return status;
+    return status;
 }
 
 bool UsdUndoRenameCommand::renameUndo()
 {
-    bool status{false};
-    {
-        // Regardless of where the edit target is currently set, switch to the
-        // layer where we copied the source prim into the destination, then
-        // restore the edit target.
-        UsdEditContext ctx(_stage, _layer);
-        status = _stage->RemovePrim(_usdDstPath);
-    }
+    // Copy the source path using CopySpec, and remove the source.
+    bool status = SdfCopySpec(_layer, _usdDstPath, _layer, _usdSrcPath);
+
     if (status) {
-        auto srcPrim = _stage->GetPrimAtPath(_usdSrcPath);
-#ifdef UFE_V2_FEATURES_AVAILABLE
-        UFE_ASSERT_MSG(srcPrim, "Invalid prim cannot be activated.");
-#else
-        assert(srcPrim);
-#endif
-        status = srcPrim.SetActive(true);
+        // remove all scene description for the given path and 
+        // its subtree in the current UsdEditTarget 
+        {
+            UsdEditContext ctx(_stage, _layer);
+            status = _stage->RemovePrim(_usdDstPath);
+        }
 
         if (status) {
-            Ufe::ObjectRename notification(_ufeSrcItem, _ufeDstItem->path());
-            Ufe::Scene::notifyObjectPathChange(notification);
+            // create a new prim at _usdSrcPath
+            auto newPrim = _stage->DefinePrim(_usdSrcPath);
+            
+            #ifdef UFE_V2_FEATURES_AVAILABLE
+                UFE_ASSERT_MSG(newPrim, "Invalid prim cannot be inactivated.");
+            #else
+                assert(newPrim);
+            #endif
+
+            // I shouldn't have to again create a sibling sceneItem here since we already have a valid _ufeSrcItem
+            // however, I get random crashes if I don't which needs furthur investigation.  HS, 6-May-2020.
+            _ufeSrcItem = createSiblingSceneItem(_ufeDstItem->path(), _usdSrcPath.GetElementString());
+
+            sendRenameNotification(_ufeSrcItem, _ufeDstItem->path());
+
             _ufeDstItem = nullptr;
         }
-	}
+    }
     else {
-        UFE_LOG(std::string("Warning: RemovePrim(") +
+        UFE_LOG(std::string("Warning: SdfCopySpec(") +
                 _usdDstPath.GetString() + std::string(") failed."));
     }
```

Comment:
On Undo, we do an opposite of Redo:

1- Call SdfCopySpec to copy ALL the child specs from _usdDstPath to _usdSrcPath. This is a crucial call.
2- Remove the prim at the _usdDstPath.
3- Create a new prim from _usdSrcPath and check if have a valid object to continue.
4- set _ufeSrcItem by calling createSiblingSceneItem ( read comments in code )
5- send rename notification
6- set _ufeDstItem to nullptr